tests/conftest.py

```diff
@@ -84,3 +84,17 @@ def many_unresolvable_items():
             )
         )
     return results
+
+
+def pytest_addoption(parser):
+    parser.addoption("--mi-version", action="store", default=None)
```

Comment:
I realized that using the MI version provided by the client to figure out which tests to run was a circular dependency. It would be possible for a bug in how we parse the MI version to cause different tests to run, which would obscure the problem with finding the MI version.

Instead, we now accept the version as an argument which can be provided at runtime.